state_machine/applications/system-check/tests/logging_test.py

```diff
@@ -0,0 +1,86 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+SD Card Logging Test
+* Author(s): Yashika Batra
+"""
+
+import os
+
+
+def sd_test():
+    """
+    Run a basic test; create a file and test existence, write to and
+    read from a file, delete file and test existence. Return result_dict
+    values accordingly
+    """
+
+    # create filepaths
+    filepath = "test.txt"
+    filepath_directory = "/sd/test.txt"
+
+    # try to create a file with the test filepath
+    try:
+        test_file = open(filepath_directory, "x")
+    except OSError:
+        # if the file already exists, remove it and create a new file
+        os.remove(filepath_directory)
+        test_file = open(filepath_directory, "x")
+
+    print("Directory after test.txt was created: {}".format(
+          os.listdir("/sd/")))
+    if filepath not in os.listdir("/sd/"):
+        return ("File creation failed.", False)
+
+    # write to file
+    test_string = "Hello World! This is a test file.\n"
+    try:
+        test_file.write(test_string)
+    except OSError as e:
+        print("Unable to write to file. {}".format(e))
+        return ("Unable to write to file. {}".format(e), False)
+    test_file.close()
+
```

Comment:
I'm not sure if it is possible but it would be nice to power cycle the SD card and reset the MCU at this point. Probably a separate test since that would halt the code execution. But, it would be a good way to ensure we're actually testing the SD card and not some file buffer in memory. If you can think about how we could do this for a little a bit that would be great. Like I said, I'm not sure if that is necessary here.

Reply:
To clarify, do we want to add a sanity check to make sure the SD card doesn't get wiped if we reset the MCU? If that's the case, we can just have a test file permanently on the SD card, and check that this file is on the card (and contains the correct information) every time we run the check script. Unplugging and replugging the board over different uses resets the MCU, so we won't need to write a separate test that resets it for us. Let me know what you think!

Reply:
I am actually interested in checking that the write actually occurs. It is pretty common in file systems for data to be held in memory and then written in a batch at some later point. From the perspective of the software, the file was written (it can be read, but it isn't actually stored on the SD card). I don't think that is what is happening here. And I think my idea is too complicated. I'll add it as an issue and then we can move on for now.